src/ampycloud/layer.py

```diff
@@ -173,23 +176,27 @@ def ncomp_from_gmm(vals : np.ndarray,
         `<https://www.astroml.org/book_figures/chapter4/fig_GMM_1D.html>`_
     """
 
-
     # If I get a 1-D array, deal with it.
     if np.ndim(vals) == 1:
         vals = vals.reshape(-1, 1)
 
     # Keep track of the original values
     vals_orig = copy.deepcopy(vals)
 
+    # If all the points are the same, I should not bother doing anything ...
+    if len(np.unique(vals_orig)) == 1:
+        logger.debug('Skipping the GMM computation: all the values are the same.')
```

Comment:
What if all values are the same but one or two? In the case of one different value it would not be possible to compute the standard deviation for that potential layer. In both cases, we should find only one layer though.

Reply:
The code works thanks to the `min_sep` keyword that ensures that close-by-points do not get split-up. It does trigger warnings, though. Which I leave untouched.